labs-ios-starter/Surveys/View Controllers/TopicNameViewController.swift

```diff
@@ -5,19 +5,79 @@ import UIKit
 
 class TopicNameViewController: UIViewController {
     // MARK: - Outlets -
-
     @IBOutlet var nameTextField: UITextField!
+    @IBOutlet var contextPicker: UIPickerView!
 
     // MARK: - Properties -
+    let topicController = TopicController()
+    let fetchController = FetchController()
+    let spinner = UIActivityIndicatorView()
+
+    var contexts: [ContextObject]? {
+        didSet {
+            setupContextPicker()
+        }
+    }
 
     // MARK: - View Lifecycle -
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        setupSpinner()
+        getAllContextQuestions()
+        self.contextPicker.delegate = self
+        self.contextPicker.dataSource = self
```

Comment:
You could set these in the storyboard if you preferred to have 2 fewer lines of code

Reply:
true. I like them here for visibility. I usually debug these first when I'm having a problem with a tableView/collectionView so makes it easier for me to see at a glance that they're set